src/pybind/mgr/cephadm/module.py

```diff
@@ -2375,6 +2426,137 @@ def upgrade_stop(self):
         self.event.set()
         return trivial_result('Stopped upgrade to %s' % target_name)
 
+    def save_spec(self, spec: ServiceSpec) -> None:
+        """
+        Attempts to save a ServiceSpec. Raises `OrchestratorError` if
+        an entry with the identifier already exist.
+
+        There are not two ways to manipulate the service_specs stored in the mon_store
+        1) Using a global definition (i.e. `ceph orch apply -i <specs_file.yaml>`)
+            This will usually contain multiple definitions of services and daemons.
+
+        2) Using the CLI for specific service types (i.e. `ceph orch apply rgw 3 --realm foo --zone bar`)
+
+        If these entries interfere, one would overwrite the other and vice versa.
+        Having the mentioned safeguard in place will prevent overwrites from two different sources.
+        """
+        json_spec = spec.to_json()
+        store = PersistentStoreDict(self, f'service_spec/{spec.service_type}')
+        name: str = spec.name or spec.service_type
+        try:
+            _ = store[name]
+            raise orchestrator.OrchestratorError(f"Specification for {name} already exists. "
+                                                 "Please review your existing specs with "
+                                                 "`ceph orch servicespecs ls` and try again.")
```

Comment:
how does this work when applying a new placement spec to an already existing service?

Reply:
That would work.
Let's say you have one RGW with `realm1` and `zone1` configured. This would be saved in:

`mgr/cephadm/service_spec/rgw/realm1.zone1/<spec>`

If add a second spec with `realm2` and `zone2` it will not interfere.